test/unit/lib/entities/context/DutchQuoteContext.test.ts

```diff
@@ -631,4 +636,42 @@ describe('DutchQuoteContext', () => {
       expect(context.hasSyntheticEligibleTokens()).toEqual(false);
     });
   });
+
+  describe('needsRouteToNative', () => {
+    it('needsRouteToNative is true if not native and not wrapped native tokenOut', async () => {
+      const request = makeDutchRequest(
+        {},
+        { useSyntheticQuotes: true },
+        { ...BASE_REQUEST_INFO_EXACT_IN, tokenOut: USDC_ADDRESS }
+      );
+      const context = new DutchQuoteContext(logger, request, provider);
+      context.dependencies();
+      expect(context.needsRouteToNative).toEqual(true);
+    });
+
+    it('needsRouteToNative is false if native tokenOut', async () => {
+      const request = makeDutchRequest(
+        {},
+        { useSyntheticQuotes: true },
+        { ...BASE_REQUEST_INFO_EXACT_IN, tokenOut: NATIVE_ADDRESS }
+      );
+      const context = new DutchQuoteContext(logger, request, provider);
+      context.dependencies();
+      expect(context.needsRouteToNative).toEqual(false);
+    });
+
+    it('needsRouteToNative is false if wrapped native tokenOut', async () => {
```

Comment:
tbh would probably be best to have a test that actually runs through the resolve logic too

Reply:
Added resolving to each of these test cases 👍